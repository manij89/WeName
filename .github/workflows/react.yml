name: React Test
on: [push]
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Install dependencies for client
        run: npm ci --prefix client
      - name: Run tests for client
        run: npm test --prefix client
 