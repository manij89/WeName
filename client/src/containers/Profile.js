import React, { useEffect } from 'react';
import Header from '../components/Header';
import { connect } from 'react-redux';
import axios from 'axios';
const BASE_URL = 'http://localhost:4002';

function Profile({ user, partner, matches }) {



  function getCode() {
    if(!user.data.linkingCode) {
      axios
      .post(`${BASE_URL}/user/${user.data.id}/code`)
      .then(res => res.status <= 400 ? res : Promise.reject(res))
      .then((res) => console.log(res))
      .catch(err => console.error(err))
    }

  }

  getCode()

  return (
    <>
      <Header />
      <h1 className='logo'>WENAME</h1>
      <div className='profile-content'>
        <p>
          Can't wait to find the perfect name with your partner?
          Invite them now!
        </p>

        <p>
          Already have a code?
          Link up now!
        </p>

      </div>
    </>
  )
}

const mapStateToProps = (state) => ({
  loading: state.loading,
  user: state.user,
  partner: state.partner,
  matches: state.matches
})

const mapDispatchToProps = (dispatch) => ({
  // setLoading: (status) => dispatch(setLoading(status)),
  // setMatches: (matches) => dispatch(setMatches(matches)),
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Profile);