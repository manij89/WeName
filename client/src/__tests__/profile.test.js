import React from 'react';
import { cleanup, render, screen } from '@testing-library/react';

import Profile from '../containers/Home'

describe('Profile', () => {
  afterEach(() => {
    cleanup();
  });

  test('render correctly', () => {
    const profile = render(<Profile />);
    expect(profile.baseElement).toMatchSnapshot();
  });

});