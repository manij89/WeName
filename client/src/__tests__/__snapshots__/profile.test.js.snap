// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile render correctly 1`] = `
<body>
  <div>
    <div>
      <div
        class="home"
      >
        <div
          class="logo"
        >
          <div>
            <span
              style="display: inline-block; width: auto;"
            >
              <span
                style="animation-name: textAnim_verticalFadeIn_1694; animation-duration: 0.8571428571428571s; animation-iteration-count: 1; animation-delay: 0s; animation-fill-mode: both; display: inline-block;"
              >
                W
              </span>
            </span>
            <span
              style="display: inline-block; width: auto;"
            >
              <span
                style="animation-name: textAnim_verticalFadeIn_1694; animation-duration: 0.8571428571428571s; animation-iteration-count: 1; animation-delay: 0.14285714285714285s; animation-fill-mode: both; display: inline-block;"
              >
                E
              </span>
            </span>
            <span
              style="display: inline-block; width: auto;"
            >
              <span
                style="animation-name: textAnim_verticalFadeIn_1694; animation-duration: 0.8571428571428571s; animation-iteration-count: 1; animation-delay: 0.2857142857142857s; animation-fill-mode: both; display: inline-block;"
              >
                N
              </span>
            </span>
            <span
              style="display: inline-block; width: auto;"
            >
              <span
                style="animation-name: textAnim_verticalFadeIn_1694; animation-duration: 0.8571428571428571s; animation-iteration-count: 1; animation-delay: 0.42857142857142855s; animation-fill-mode: both; display: inline-block;"
              >
                A
              </span>
            </span>
            <span
              style="display: inline-block; width: auto;"
            >
              <span
                style="animation-name: textAnim_verticalFadeIn_1694; animation-duration: 0.8571428571428571s; animation-iteration-count: 1; animation-delay: 0.5714285714285714s; animation-fill-mode: both; display: inline-block;"
              >
                M
              </span>
            </span>
            <span
              style="display: inline-block; width: auto;"
            >
              <span
                style="animation-name: textAnim_verticalFadeIn_1694; animation-duration: 0.8571428571428571s; animation-iteration-count: 1; animation-delay: 0.7142857142857142s; animation-fill-mode: both; display: inline-block;"
              >
                E
              </span>
            </span>
          </div>
        </div>
        <button
          class="home_button"
        >
          Start
        </button>
        <img
          alt="stork"
          class="stork"
          src="stork.svg"
        />
        <img
          alt="grass"
          class="grass"
          src="grass.svg"
        />
      </div>
      <div
        class="info-container"
      >
        <div
          class="MuiPaper-root info-card MuiPaper-elevation7 MuiPaper-rounded"
        >
          <h1>
            Find your favorite baby names
          </h1>
          <ul
            class="MuiList-root MuiList-padding"
          >
            <div>
              <li
                class="MuiListItem-root MuiListItem-gutters"
              >
                <div
                  class="MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    Register and go to your "Account"
                  </span>
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters"
              >
                <div
                  class="MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    Share or enter the personal code to play with your partner
                  </span>
                </div>
              </li>
              <li
                class="MuiListItem-root MuiListItem-gutters"
              >
                <div
                  class="MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    Keep swiping and check your matches
                  </span>
                </div>
              </li>
            </div>
          </ul>
          <div>
            <a
              href="/register"
            >
              <button
                class="info-button"
              >
                Sign up
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
